code/main/consent/consentState.brs

```diff
@@ -0,0 +1,139 @@
+                _adb_logError("ConsentState::setCollectConsent() - Failed to set collect consent value, invalid collect consent value: (" + FormatJson(collectConsent) + ") passed.")
+                return
+            end if
+
+            m._saveCollectConsent(collectConsent)
+        end sub,
+
+        getCollectConsent: function() as dynamic
+            _adb_logVerbose("ConsentState::getCollectConsent() - Getting collect consent value.")
+            if m._collectConsent = invalid
+                _adb_logVerbose("ConsentState::getCollectConsent() - Collect consent value not found in memory. Loading from persistence.")
+                m._collectConsent = m._loadCollectConsent()
+            end if
+
+            if m._collectConsent = invalid
+                _adb_logVerbose("ConsentState::getCollectConsent() - Collect consent value not found in persistence. Extracting from configuration.")
+                m._collectConsent = m._getCollectConsentFromConfiguration()
+            end if
+
+            _adb_logVerbose("ConsentState::getCollectConsent() - Returning collect consent value: (" + FormatJson(m._collectConsent) + ")")
+            return m._collectConsent
+        end function
+
+        _loadCollectConsent: function() as dynamic
+            _adb_logVerbose("ConsentState::_loadCollectConsent() - Loading collect consent value from persistence.")
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            collectConsentValue = localDataStoreService.readValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT)
+
+            return collectConsentValue
+        end function,
+
+        _saveCollectConsent: function(collectConsentValue as dynamic) as void
+            if not m._isValidConsentValue(collectConsentValue) then
+                _adb_logError("ConsentState::_saveCollectConsent() - Invalid collect consent value: (" + FormatJson(collectConsentValue) + ").")
+                return
+            end if
+
+            ''' cache the collect consent value
+            m._collectConsent = collectConsentValue
+
+            _adb_logVerbose("ConsentState::_saveCollectConsent() - Saving collect consent value " + FormatJson(collectConsentValue) + " in persistence.")
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            localDataStoreService.writeValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT, collectConsentValue)
+        end function,
+
+        _deleteCollectConsent: function() as void
+            _adb_logVerbose("ConsentState::_deleteCollectConsent() - Deleting collect consent value from persistence.")
+
+            m._collectConsent = invalid
+
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            localDataStoreService.removeValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT)
+            return
+        end function,
```

Comment:
How/when is `_deleteCollectConsent` being used? In iOS/Android we don't have the functionality to remove consent values, just to update them.

Reply:
Yes, we might not need this. Will delete it later if I don't see any usecase.